src/Take.Blip.Builder/Actions/ProcessHttp/ProcessHttpAction.cs

```diff
@@ -140,10 +133,17 @@ private void AddBotIdentityToHeaders(HttpRequestMessage httpRequestMessage, ICon
             }
         }
 
-        private void AddHeadersToCommandRequest(HttpRequestMessage httpRequestMessage, string currentStateId, string ownerIdentity, string uri)
-        {      
-             httpRequestMessage.Headers.Add(Constants.BLIP_BOT_HEADER, ownerIdentity);
-             httpRequestMessage.Headers.Add(Constants.BLIP_STATEID_HEADER, currentStateId);
+        private void AddHeadersToCommandRequest(HttpRequestMessage httpRequestMessage, string currentStateId, string ownerIdentity, string absoluteUri)
+        {
+            var uriList = _configuration.InternalUris.Split(";");
+            foreach (var uri in uriList)
+            {
+                if (absoluteUri.Contains(uri))
+                {
+                    httpRequestMessage.Headers.Add(Constants.BLIP_BOT_HEADER, ownerIdentity);
```

Comment:
the headers will be duplicated if two uris were contained in absoluteUris

Reply:
ok